cron/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sort"
+	"time"
+
+	"github.com/jszwec/csvutil"
+)
+
+type Repository struct {
+	Repo     string `csv:"repo"`
+	Metadata string `csv:"metadata,omitempty"`
+}
+
+func main() {
+	fileName := fmt.Sprintf("%d-%d-%d.json", time.Now().Month(), time.Now().Day(), time.Now().Year())
+	result, err := os.Create(fileName)
+	if err != nil {
+		panic(err)
+	}
+	defer result.Close()
+	projects, err := os.OpenFile(os.Args[1], os.O_RDONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer projects.Close()
+
+	repos := []Repository{}
+	data, err := ioutil.ReadAll(projects)
+	if err != nil {
+		panic(err)
+	}
+	err = csvutil.Unmarshal(data, &repos)
+	if err != nil {
+		panic(err)
+	}
+	sort.Slice(repos, func(i, j int) bool {
+		return repos[i].Repo < repos[j].Repo
+	})
+
+	//nolint
+	const checks string = "--checks=Active,CI-Tests,CII-Best-Practices,Code-Review,Contributors,Frozen-Deps,Fuzzing,Packaging,Pull-Requests,SAST,Security-Policy,Signed-Releases,Signed-Tags"
+
+	if _, err = result.Write([]byte(`{"results":[`)); err != nil {
+		panic(err)
+	}
+	for _, r := range repos {
+		fmt.Println(r.Repo)
+		//nolint
+		cmd := exec.Command("../scorecard", fmt.Sprintf("--repo=%s", r.Repo), fmt.Sprintf("--metadata=%s", r.Metadata), checks, "--show-details", "--format=json")
+		// passing the external github token the cmd
+		cmd.Env = append(cmd.Env, fmt.Sprintf("GITHUB_AUTH_TOKEN=%s", os.Getenv("GITHUB_AUTH_TOKEN")))
+		cmd.Stderr = io.Writer(os.Stdout)
+
+		// Execute the command
+		data, err := cmd.Output()
+		if err != nil {
+			fmt.Printf("error:failed for the repo %s and the error is %s", r.Repo, err.Error())
+			// continuing because this is just for that repo that failed.
+			continue
+		}
+
+		_, err = result.WriteString(fmt.Sprintf("\n%s,", string(data)))
+		if err != nil {
+			panic(err)
+		}
+	}
+	result.Close()
+
+	// sed -i '$d' "$output" # removing the trailing comma which will be last line.
+	if err := exec.Command("sed", "-i", "$d", fileName).Run(); err != nil {
```

Comment:
This can be all much simplified by handling this in the loop above rather than trying to emulate the bash script. We can also avoid re-opening the file.

E.g. something like

```go
for i, r := range repos {
  ...

  if i < len(repos) - 1 {
    result.WriteString(",")
  }
  ...
}


result.WriteString("\n]}\n")
```

Reply:
Nice! Agreed.